tools/restatectl/src/connection.rs

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{cmp::Ordering, collections::HashMap, fmt::Display, future::Future, sync::Arc};
+
+use cling::{prelude::Parser, Collect};
+use itertools::{Either, Itertools};
+use rand::{seq::SliceRandom, thread_rng};
+use tokio::sync::{Mutex, MutexGuard};
+use tonic::{codec::CompressionEncoding, transport::Channel, Response, Status};
+use tracing::debug;
+
+use restate_core::protobuf::node_ctl_svc::{
+    node_ctl_svc_client::NodeCtlSvcClient, GetMetadataRequest, IdentResponse,
+};
+use restate_types::{
+    logs::metadata::Logs,
+    net::AdvertisedAddress,
+    nodes_config::{NodesConfiguration, Role},
+    protobuf::common::{MetadataKind, NodeStatus},
+    storage::{StorageCodec, StorageDecode, StorageDecodeError},
+    Version, Versioned,
+};
+
+use crate::util::grpc_channel;
+
+#[derive(Clone, Parser, Collect, Debug)]
+pub struct ConnectionInfo {
+    /// Specify the node address. This option can be used multiple times.
+    #[clap(
+        long="address",
+        value_hint = clap::ValueHint::Url,
+        default_value = "http://localhost:5122/",
+        env = "RESTATE_ADDRESS",
+        global = true
+    )]
+    pub addresses: Vec<AdvertisedAddress>,
+
+    /// Sync metadata from metadata store first
```

Comment:
Maybe make this a bit more explanatory what this means for users and when they should use it vs. when not.